src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,14 +6,48 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int arraylength = 0;
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                if (i == j) {
+                    arraylength++;
+                }
+            }
+        }
+        int[] array = new int[arraylength];
+        for (int j = 0; j < matrix.length; j++) {
+            for (int k = 0; k < matrix[0].length; k++) {
+                if (j == k) {
+                    array[j] = matrix[j][k];
+                }
+            }
+        }
+        return array;
     }
 
+
     public int getMin(int[][] matrix) {
-        return 0;
+        int minvalue = matrix[0][0];
```

Comment:
```suggestion
        int minValue = matrix[0][0];
```
remember about camelCase